DirSearcherClient/Program.cs

```diff
@@ -181,23 +178,29 @@ static AuthenticationResult GetToken(string tenant)
             {
                 result = ctx.AcquireTokenSilentAsync(resource, clientId).Result;
             }
+            catch(AggregateException ex) when (ex.InnerException is AdalSilentTokenAcquisitionException)
+            {
+                result = GetTokenViaCode(ctx);
+            }
+            catch (AdalSilentTokenAcquisitionException)
```

Comment:
>(AdalSilentTokenAcquisitionException) [](start = 18, length = 37)

Minor... combine with catch above ? catch(ex) when (ex is ...) || (ex is ...) ? can e.g. be done via an exception filtering method (private bool IsAdalSilentTokenAquisitionException(exception e) {}) ... not sure though if this works for .net core...

Reply:
yes, I agree.